webap/static/seechange.js

```diff
@@ -0,0 +1,587 @@
+import { rkWebUtil } from "./rkwebutil.js";
+
+// Namespace, which is the only thing exported
+
+var seechange = {};
+
+seechange.nullorfixed = function( val, num ) { return val == null ? null : val.toFixed(num); }
+
+// **********************************************************************
+// **********************************************************************
+// **********************************************************************
+// The global context
+
+seechange.Context = function()
+{
+    this.parentdiv = document.getElementById( "pagebody" );
+    let h1 = rkWebUtil.elemaker( "h1", this.parentdiv, { "text": "SeeChange Webap" } );
+    this.maindiv = rkWebUtil.elemaker( "div", this.parentdiv );
+    this.frontpagediv = null;
+
+    // TODO : make this configurable?  Or at least remember how to
+    //   detect in javascript what the URL you're running from is.  (In
+    //   case the webap is not running as the root ap of the webserver.)
+    this.connector = new rkWebUtil.Connector( "/" );
+}
+
+seechange.Context.prototype.render_page = function()
+{
+    var self = this;
```

Comment:
I don't think you used `self` anywhere...

Reply:
This is a kneejerk thing I'll often do.  I find the javascript scoping rules for "this" nonintuitive (and based on web searches, lots of other people do to).  So often I'm putting in callbacks and stuff that need access to the "this" of the function where I'm building the callback (which will not be the "this" in the fucntion that gets called back) that I often stick this in when I start a function.

(Of course, just as often, I forget to put this in, and get an "undefined self" when I try to run it later.)

Reply:
Looking it, I *do* use self here -- it's in the callbacks to the document elements, e.g.:

`        let button = rkWebUtil.button( p, "Show Exposures", function() { self.show_exposures(); } );`